tests/test_deprecated_trainer.py

```diff
@@ -0,0 +1,412 @@
+from unittest import TestCase
+
+import evaluate
+import pytest
+from datasets import Dataset
+from sentence_transformers import losses
+from transformers.testing_utils import require_optuna
+from transformers.utils.hp_naming import TrialShortNamer
+
+from setfit import logging
+from setfit.modeling import SetFitModel, SupConLoss
+from setfit.trainer import SetFitTrainer
+from setfit.utils import BestRun
+
+
+logging.set_verbosity_warning()
+logging.enable_propagation()
+
+
+class SetFitTrainerTest(TestCase):
```

Comment:
Note to our future selves - at some point we should refactor all the tests to follow the `unittest.TestCase` grouping instead of single functions. Personally, I think this makes it easier to understand what parts of the code are being tested :)

Reply:
Noted! Beyond that, via `pytest` you can order specific classes to be tested, even if a file has multiple of these classes. Can definitely help speed up development if you don't have to wait for additional unrelated tests to pass.